test/cpp/interop/stress_test.cc

```diff
@@ -199,23 +214,48 @@ int main(int argc, char** argv) {
   LogParameterInfo(server_addresses, tests);
 
   WeightedRandomTestSelector test_selector(tests);
+  MetricsServiceImpl metrics_service;
 
   gpr_log(GPR_INFO, "Starting test(s)..");
 
   vector<thread> test_threads;
+
   int thread_idx = 0;
   for (auto it = server_addresses.begin(); it != server_addresses.end(); it++) {
-    StressTestInteropClient* client = new StressTestInteropClient(
-        ++thread_idx, *it, test_selector, FLAGS_test_duration_secs,
-        FLAGS_sleep_duration_ms);
-
-    test_threads.emplace_back(
-        thread(&StressTestInteropClient::MainLoop, client));
+    // TODO(sreek): This will change once we add support for other tests
+    // that won't work with InsecureCredentials()
+    std::shared_ptr<grpc::Channel> channel(
```

Comment:
This is because multiple StressTestInteropClient objects might be referring to the same channel (based on the FLAGS_num_stubs_per_channel).